src/common/abstractPromote.ts

```diff
@@ -101,6 +102,52 @@ export abstract class PromoteCommand<T extends typeof SfCommand> extends SfComma
     };
   }
 
+  /**
+   *
+   * Sends an Http request to the target org to initiate a promotion.
+   *
+   * @param doceOrg target org.
+   * @returns Aor Id of promote operation.
+   */
+  protected async requestPromotionFlow(doceOrg: Org): Promise<string> {
+    try {
+      return await this.requestPromotion(doceOrg);
+    } catch (error) {
+      const err = error as ApiError;
+      // if we get a 409 error then call the retry flow
+      if (err.errorCode === HTTP_CONFLICT_CODE) {
+        this.spinner.start('Synchronization of source control system events in progress');
+        return await this.retryRequestPromotion(doceOrg, this.numRetries409);
+      }
+      throw error;
+    } finally {
+      this.spinner.stop();
```

Comment:
```suggestion
      if (spinnerStarted) this.spinner.stop();
```

If we get an exception (not 409) in the first call, this will call this.spinner.stop() without it being started...right?  I am not sure if that is bad or not, but you can guard against it.